bitswap/client/internal/sessioninterestmanager/sessioninterestmanager.go

```diff
@@ -177,3 +177,26 @@ func (sim *SessionInterestManager) InterestedSessions(keySets ...[]cid.Cid) []ui
 	}
 	return ses
 }
+
+// Filters only the keys that are wanted by at least one session
+func (sim *SessionInterestManager) FilterInterests(keySets ...[]cid.Cid) [][]cid.Cid {
+	sim.lk.RLock()
+	defer sim.lk.RUnlock()
+
+	result := make([][]cid.Cid, len(keySets))
```

Comment:
Instead of creating a new slice, modify in place.
```suggestion
	result := keySets[:0]
```